packages/sku/src/utils/detectUnnecessaryPolyfillsFromDependencies.ts

```diff
@@ -0,0 +1,58 @@
+import { readFileSync, existsSync } from 'node:fs';
+import { getPathFromCwd } from '@sku-lib/utils';
+import {
+  type DetectedPolyfillWithSource,
+  getDeprecatedPolyfill,
+} from './polyfillDetector.js';
+
+/**
+ * Detects unnecessary polyfills from package.json dependencies
+ * by matching package names against the registry.
+ */
+export const detectUnnecessaryPolyfillsFromDependencies =
+  (): DetectedPolyfillWithSource[] => {
+    const packageJsonPath = getPathFromCwd('package.json');
+
+    if (!existsSync(packageJsonPath)) {
+      throw new Error(
+        `package.json not found at ${packageJsonPath} - this is required for polyfill detection`,
+      );
+    }
+
+    try {
+      const packageJsonContent = readFileSync(packageJsonPath, 'utf-8');
+      const packageJson = JSON.parse(packageJsonContent);
```

Comment:
It's generally not recommended to check for file existence before reading the file because of race conditions. It's not a big deal here, but readFile will throw an error anyway, and you can check the error code (ENOENT I think) if you want to display a specific error message. 


```suggestion
    const packageJsonPath = getPathFromCwd('package.json');

    try {
      const packageJsonContent = readFileSync(packageJsonPath, 'utf-8');
      const packageJson = JSON.parse(packageJsonContent);
```